### ISSUE TITLE
Question: Vitess on ARM 

### ISSUE THREAD

**Issue by timkang85** [vitessio/vitess#9797]
### Question

```markdown
I have a question on how to build arm vitess image, and if is there any caveat to use vitess on arm? 

Also, I've found that etcd on arm is unstable, is this still?
```

**Comment by frouioui** [comment]
I believe @vmg is building/running Vitess on an ARM CPU.

The version `3.5` of etcd is supported but considered as unstable on Linux ARM machines:
> _etcd is guaranteed to build, may be lightly tested (or not), and so it should be considered unstable._
(https://etcd.io/docs/v3.5/op-guide/supported-platform/)

**Comment by vmg** [comment]
Nope, not at all. I don't own any ARM CPUs. Vitess isn't tested on ARM on CI either. Running ARM Vitess in production seems like a risky proposition at this point, since nobody is doing it.

@timkang85: if you have a specific interest on running Vitess on ARM, you'll probably need to contribute changes to the project, as bugs will surely be found. The ideal first step would be a CI pass (maybe just for the unit tests) that runs on ARM.

**Comment by derekperkins** [comment]
Following up here, etcd now has first class support for ARM64, and now Percona has added support.
https://www.percona.com/blog/percona-server-for-mysql-and-percona-xtrabackup-now-available-for-arm64/

**Comment by martin-g** [comment]
As a CNCF project you could use `runs-on: openeuler-6cpu-14gb` in your Github Actions workflows. The jobs will be executed on ARM64 VMs with openEuler Linux distro (YUM/DNF based, like RHEL).

**Comment by ranimandepudi** [comment]
Hi all,

I'm an engineer at Arm, and we're enthusiastic about enhancing Arm64 support within the Vitess project. I appreciate the insights shared in this thread and would like to provide some updates and propose collaborative steps forward.

**etcd ARM64 Support:**

Regarding concerns about etcd's stability on Arm64, it's worth noting that etcd has made significant strides in this area. As of version 3.5, etcd has achieved Tier 1 support for Arm64, indicating full support with successful completion of all tests, including functional and robustness tests. This development enhances the feasibility of running Vitess on Arm64 platforms.

**Proposed Collaborative Steps:**

To further Arm64 support in Vitess, I am planning to work on following actions:

**Build and Test Vitess on Arm64:**

- Compile Vitess on ARM64 platforms.
- Run the complete test suite to identify any architecture-specific issues.

**Develop Arm64 Docker Images:**

-  Utilize docker buildx or build to create multi-architecture images.
- Ensure these images are tested for Arm64 environment and current environments.

If everything sounds cool then :

**Integrate ARM64 into CI/CD Pipelines:**

- Set up continuous integration workflows to build and test Vitess on Arm64.
- Monitor these workflows to maintain stability and performance.

**Community Collaboration:**

- Engage with the Vitess community to share findings and gather feedback.
- Coordinate efforts to address any Arm64-specific challenges.

I'm committed to contributing to these initiatives and collaborating closely with the community.
